compiler/rustc_parse/src/parser/generics.rs

```diff
@@ -275,6 +275,11 @@ impl<'a> Parser<'a> {
             self.expect_gt_or_maybe_suggest_closing_generics(&params)?;
             (params, span_lo.to(self.prev_token.span))
         } else {
+            if self.token == token::PathSep && self.look_ahead(1, |t| *t == TokenKind::Lt) {
```

Comment:
This is still not recovering the list of generics correctly. Please actually `eat` the `::` token so that we don't return an *empty* list of params.

This also should be gated behind `self.may_recover()`.

Reply:
Specifically, this should be moved *before* the `eat_lt` call above.

Reply:
> Specifically, this should be moved _before_ the `eat_lt` call above.

Thank you, please help me see if there is anything else I need to modify.